src/util.rs

```diff
@@ -28,7 +28,7 @@ where
 {
     let mut contents = String::new();
     let mut file = BufReader::new(File::open(file)
-        .internal_error("util", "failed to open file")?);
+        .internal_error("util", &("failed to open file".to_owned() + file))?);
```

Comment:
Is it possible for you to use the same `format!` approach used elsewhere in the project? Using `+` for string concatenation is not very idiomatic.

Reply:
Sure, I did this in https://github.com/greshake/i3status-rust/pull/394/commits/3cbfae5bb5a9981fe43d436c57a28c0e93ed5fcb.